bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/SortedLedgerStorage.java

```diff
@@ -56,18 +63,22 @@
 
     EntryMemTable memTable;
     private ScheduledExecutorService scheduler;
-    private StateManager stateManager;
     private ServerConfiguration conf;
     private StatsLogger statsLogger;
-    private final InterleavedLedgerStorage interleavedLedgerStorage;
+    protected final List<InterleavedLedgerStorage> interleavedLedgerStorageList;
+    private int numberOfDirs;
+    private String interleavedLedgerStorageClazz = InterleavedLedgerStorage.class.getName();
 
     public SortedLedgerStorage() {
-        this(new InterleavedLedgerStorage());
+        this(null);
     }
 
     @VisibleForTesting
-    protected SortedLedgerStorage(InterleavedLedgerStorage ils) {
-        interleavedLedgerStorage = ils;
+    protected SortedLedgerStorage(String interleavedLedgerStorageClazz) {
```

Comment:
Why do we pass a class name here?

Reply:
this is for InterleavedLedgerStorage's child class: SlowInterleavedLedgerStorage

we can init SortedLedgerStorage with overide the diff child InterleavedLedgerStorage

<img width="1056" alt="image" src="https://user-images.githubusercontent.com/42990025/190967167-61cc57cc-172b-433a-b93c-928f57645883.png">